packages/processors/src/strategy/donationVotingMerkleDistributionDirectTransfer/handlers/allocated.handler.ts

```diff
@@ -0,0 +1,198 @@
+        const { srcAddress } = this.event;
+        const { recipientId: _recipientId, amount, token: _token } = this.event.params;
+
+        if (!this.event.params.origin) {
+            throw new OriginMissing();
+        }
+
+        const round = await this.getRoundOrThrow(srcAddress);
+        const application = await this.getApplicationOrThrow(round.id, _recipientId);
+
+        const donationId = this.getDonationId(this.event.blockNumber, this.event.logIndex);
+
+        const token = this.getTokenOrThrow(_token);
+        const matchToken = this.getTokenOrThrow(round.matchTokenAddress);
+
+        const { amountInUsd, timestamp: priceTimestamp } = await getTokenAmountInUsd(
+            this.dependencies.pricingProvider,
+            token,
+            amount,
+            this.event.blockTimestamp,
+        );
+        let amountInRoundMatchToken: bigint | null = null;
+        amountInRoundMatchToken =
+            matchToken.address === token.address
+                ? amount
+                : (
+                      await getUsdInTokenAmount(
+                          this.dependencies.pricingProvider,
+                          matchToken,
+                          amountInUsd,
+                          this.event.blockTimestamp,
+                      )
+                  ).amount;
+
+        const parsedMetadata = this.parseMetadataOrThrow(application.metadata);
+
+        const donation: Donation = {
+            id: donationId,
+            chainId: this.chainId,
+            roundId: round.id,
+            applicationId: application.id,
+            donorAddress: getAddress(this.event.params.origin),
+            recipientAddress: getAddress(parsedMetadata.application.recipient),
+            projectId: application.projectId,
+            transactionHash: this.event.transactionFields.hash,
+            blockNumber: BigInt(this.event.blockNumber),
+            tokenAddress: token.address,
+            amount: amount,
+            amountInUsd,
+            amountInRoundMatchToken,
+            timestamp: new Date(priceTimestamp), //TODO: ask Gitcoin if this is correct
+        };
+
+        return [
+            {
+                type: "InsertDonation",
+                args: { donation },
+            },
+        ];
+    }
+
+    /**
+     * Retrieves a round by its strategy address.
+     * @param {Address} strategyAddress - The address of the strategy.
+     * @returns {Promise<Round>} The round found.
+     * @throws {RoundNotFound} if the round does not exist.
+     */
+    private async getRoundOrThrow(strategyAddress: Address): Promise<Round> {
+        const normalizedStrategyAddress = getAddress(strategyAddress);
+        const round = await this.dependencies.roundRepository.getRoundByStrategyAddress(
+            this.chainId,
+            normalizedStrategyAddress,
+        );
+
+        if (!round) {
+            throw new RoundNotFound(this.chainId, normalizedStrategyAddress);
```

Comment:
this is out of scope, but in this case this should be a critical error, right? i mean the round should always be defined for the strategy

Reply:
exactly